x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rule_status_filter.tsx

```diff
@@ -34,45 +40,59 @@ export const RuleStatusFilter = (props: RuleStatusFilterProps) => {
 
   const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
 
-  const onFilterItemClick = useCallback(
-    (newOption: RuleStatus) => () => {
-      if (selectedStatuses.includes(newOption)) {
-        onChange(selectedStatuses.filter((option) => option !== newOption));
-        return;
-      }
-      onChange([...selectedStatuses, newOption]);
+  const onFilterItemChange: EuiSelectableProps['onChange'] = useCallback(
+    async (newOptions: EuiSelectableOption[]) => {
+      const selected = newOptions
+        .filter(({ checked, key }) => checked && key)
+        .map(({ key }) => key as RuleStatus);
+
+      await onChange(selected);
     },
-    [selectedStatuses, onChange]
+    [onChange]
   );
 
   const onClick = useCallback(() => {
     setIsPopoverOpen((prevIsOpen) => !prevIsOpen);
   }, [setIsPopoverOpen]);
 
-  const renderRuleStateOptions = (status: 'enabled' | 'disabled' | 'snoozed') => {
-    if (status === 'enabled') {
-      return (
-        <FormattedMessage
-          id="xpack.triggersActionsUI.sections.ruleDetails.ruleStateFilter.enabledOptionText"
-          defaultMessage="Rule is enabled"
-        />
-      );
-    } else if (status === 'disabled') {
-      return (
-        <FormattedMessage
-          id="xpack.triggersActionsUI.sections.ruleDetails.ruleStateFilter.disabledOptionText"
-          defaultMessage="Rule is disabled"
-        />
-      );
-    } else if (status === 'snoozed') {
-      return (
-        <FormattedMessage
-          id="xpack.triggersActionsUI.sections.ruleDetails.ruleStateFilter.snoozedOptionText"
-          defaultMessage="Rule has snoozed"
-        />
-      );
-    }
-  };
+  const selectableOptions = useMemo<EuiSelectableOption[]>(() => {
+    const partialOptions: Array<{ key: RuleStatus; label: string }> = [
```

Comment:
nitpick: can we move this outside of the component if it doesn't depend on any component state